### PR TITLE
Cleanup cert management in Helm charts

### PR DESCRIPTION
**Pull request by ff-kamal**
Cleanup some of the cert management around the Helm charts. This includes:

- Removing hardcoded secret names (and use the previously defined one in config)
- Add config for the endpoint of Let's Encrypt that we want to target, for testing infrastructure purposes
- Grouping cert configs together.

Note: These changes require backwards incompatible changes (as we are changing the location of some of the configuration values to group them together). If we feel like this would be an unnecessary burden, we can change back the location of the configuration values and leave the rest of the changes.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Cleanup
# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: charts/featureform/templates/ingress/issuer.yaml [hunk 1]
```diff
@@ -5,19 +5,24 @@
...
 apiVersion: cert-manager.io/v1
 kind: Issuer
 metadata:
-  name: letsencrypt-prod
+  name: letsencrypt
 spec:
   acme:
-    server: https://acme-v02.api.letsencrypt.org/directory
+    server: {{
+              ternary
```
Comment (anthonylasso) [inline charts/featureform/templates/ingress/issuer.yaml:16]: q: how do these options work? is it just a set of cert servers?

File: charts/featureform/values.yaml [hunk 2]
```diff
@@ -20,16 +20,25 @@ repository: "featureformenterprise"
...
+  # Will automatically create a self-signed cert that can be used if enabled
+  selfSignedCert: true
 
-# Note: If post publicCert and selfSignedCert are disabled, an existing certificate will need to be supplied
+  # Whether to use the prod letsencrypt endpoint or the staging. The production endpoint returns
+  # certificates that are trusted be well-known Certificate Authorities, however they have a low
+  # request limit (approximately 5 / week at the time of writing). During testing of infrastructure,
+  # this might be too low, so they also have a staging endpoint, with a much higher limit, but
+  # by default they're not trusted by any well-known CA (but you can add the certs manually).
+  # See https://letsencrypt.org/docs/staging-environment/
```
Comment (anthonylasso) [inline charts/featureform/values.yaml:35]: note: that limit restriction _might_ be ok for now. we don't normally stand up or tear down the internal box as much, not even the nightly box saw 5 times a week cycles.

**Review by anthonylasso — APPROVED** [review]

**Comment by anthonylasso** [comment]
The compatibility issue shouldn't be a big deal with new installs from the OSS repo. Anyone else upgrading will just need to take note of the option changes and edit their values file accordingly.

File: charts/featureform/templates/ingress/issuer.yaml (hunk 1 above)
Comment (ff-kamal) [reply charts/featureform/templates/ingress/issuer.yaml:16]: That's correct, it's a couple of hardcoded cert servers, with the staging one specified at https://letsencrypt.org/docs/staging-environment/. The production one is just the normal one we've been using before.

Comment (anthonylasso) [reply charts/featureform/templates/ingress/issuer.yaml:16]: Gotcha make sense.

**Pull request merged**